src/Core/src/Sts/StsClient.php

```diff
@@ -89,9 +89,121 @@ public function getCallerIdentity($input = []): GetCallerIdentityResponse
+                        0 => 'v4',
+                    ],
+                ];
+            case 'cn-north-1':
+            case 'cn-northwest-1':
+                return [
+                    'endpoint' => 'https://sts.%region%.amazonaws.com.cn',
+                    'signRegion' => $region,
+                    'signService' => 'sts',
+                    'signVersions' => [
+                        0 => 'v4',
+                    ],
+                ];
+            case 'us-gov-east-1':
+            case 'us-gov-west-1':
+                return [
+                    'endpoint' => 'https://sts.%region%.amazonaws.com',
+                    'signRegion' => $region,
+                    'signService' => 'sts',
+                    'signVersions' => [
+                        0 => 'v4',
+                    ],
+                ];
+            case 'us-iso-east-1':
+                return [
+                    'endpoint' => 'https://sts.%region%.c2s.ic.gov',
+                    'signRegion' => $region,
+                    'signService' => 'sts',
+                    'signVersions' => [
+                        0 => 'v4',
+                    ],
+                ];
+            case 'us-isob-east-1':
+                return [
+                    'endpoint' => 'https://sts.%region%.sc2s.sgov.gov',
+                    'signRegion' => $region,
+                    'signService' => 'sts',
+                    'signVersions' => [
+                        0 => 'v4',
+                    ],
+                ];
+            case 'us-east-1-fips':
+                return [
+                    'endpoint' => 'https://sts-fips.us-east-1.amazonaws.com',
+                    'signRegion' => 'us-east-1',
+                    'signService' => 'sts',
+                    'signVersions' => [
+                        0 => 'v4',
+                    ],
+                ];
+            case 'us-east-2-fips':
+                return [
+                    'endpoint' => 'https://sts-fips.us-east-2.amazonaws.com',
+                    'signRegion' => 'us-east-2',
+                    'signService' => 'sts',
+                    'signVersions' => [
+                        0 => 'v4',
+                    ],
+                ];
+            case 'us-west-1-fips':
+                return [
+                    'endpoint' => 'https://sts-fips.us-west-1.amazonaws.com',
+                    'signRegion' => 'us-west-1',
+                    'signService' => 'sts',
+                    'signVersions' => [
+                        0 => 'v4',
+                    ],
+                ];
+            case 'us-west-2-fips':
+                return [
+                    'endpoint' => 'https://sts-fips.us-west-2.amazonaws.com',
+                    'signRegion' => 'us-west-2',
+                    'signService' => 'sts',
+                    'signVersions' => [
+                        0 => 'v4',
+                    ],
+                ];
+        }
+
+        return [
```

Comment:
Should this global endpoint be used also for unsupported regions or only for `null` ?

Reply:
hmm indeed, some endpoint may have regional + global endpoint, in that case we have a flag telling if the endpoint is regionnalized and could be used to check this case

Reply:
refactored to check if the region is supported. 

this also fix a potential issue where region with typo would have been use the global endpoint.